src/MongoDB.Driver/Linq/MongoQueryable.cs

```diff
@@ -83,19 +84,19 @@ public static Task<bool> AnyAsync<TSource>(this IMongoQueryable<TSource> source,
         /// <param name="stage">The stage to append.</param>
         /// <param name="resultSerializer">The result serializer.</param>
         /// <returns>The queryable with a new stage appended.</returns>
-        public static IMongoQueryable<TResult> AppendStage<TSource, TResult>(
-            this IMongoQueryable<TSource> source,
+        public static IQueryable<TResult> AppendStage<TSource, TResult>(
+            this IQueryable<TSource> source,
             PipelineStageDefinition<TSource, TResult> stage,
             IBsonSerializer<TResult> resultSerializer = null)
         {
             Ensure.IsNotNull(source, nameof(source));
             Ensure.IsNotNull(stage, nameof(stage));
 
-            return (IMongoQueryable<TResult>)source.Provider.CreateQuery<TResult>(
+            return (IQueryable<TResult>)source.Provider.CreateQuery<TResult>(
                 Expression.Call(
                     null,
                     GetMethodInfo(AppendStage, source, stage, resultSerializer),
-                    Expression.Convert(source.Expression, typeof(IMongoQueryable<TSource>)),
+                    source.Expression,
```

Comment:
All these converts can now be removed. I'm not sure they were ever actually needed.